Civi/Core/SettingsMetadata.php

```diff
@@ -49,8 +59,30 @@ class SettingsMetadata {
    *   - help_text
    *   - options
    *   - pseudoconstant
+   *   - global_name (name for php constant / environment variable corresponding to this setting)
+   *   - load_from_env (whether this setting should be read from corresponding environment variable)
+   *   - is_constant (whether a PHP constant should be defined for this setting)
```

Comment:
Nitpick: The naming feels asymmetric. Any of these would feel more consistent:

* (A) For one string, two bools
    * (A1) `string global_name` + `bool is_constant` + `bool is_env_var`
    * (A2) `string global_name` + `bool use_constant` + `bool use_env_var`
    * (A3) `string global_name` + `bool load_constant` + `bool load_env_var`
    * (Between these, `is_*` has some precedent within setting-metadata, and it's closer to current names. `use_*` and `load_*` seem OK if they at least match each other, but fwiw I don't see as much precedent.)
* (B) For two strings
    *  (B1)`string constant' + `string env_var`
* (C) For one string, one array
    * (C1) `string global_name` + `array sources`
    * (List of supported sources -- php-constants, environment, database, php-settings, or whatever)

The smallest revision would be to go to (A1). And it would be more consistent. 😄 

(*(B) still looks cleaner imho. But it's subjective, and I don't want to block the whole changeset on that. Still, for a couple months, these will just be internal keys... so if you see the light later, we could still change it...*  😉)

(*(C) has its merits, but it's slippery to implement.*)

Reply:
So found this interesting because I don't like B much at all...

For me the `is_constant`ness is quite different from the `whether the value can load_from_env var`ness. 

The latter I thing is very tightly saying "the value for this setting *might* come from the environment, so check for it". IMHO it should definitely only be about the source of the value. `putenv()` doesn't belong anywhere in the codebase.

`is_constant` is a bit more slippery but I think really it should mean something like is "the value of this setting must not change during a given request".  It so happens that currently we use php-constants to implement this, because they behave in this way. And given there are all sorts of pre-existing places where php-constants directly  written to / read, if we're in the business of managing settings, we have to use them as both source and sink (i.e. checking for pre-existing values, and then making sure we define them for any values we know). 

That's maybe a bit abstract - in practical terms I think there may be settings which are reasonable to `load_from_env` but needn't `be_constant`. Slightly random example but take `menubar_color` -- useful to be able to override with an env var on staging, but I don't think it needs to be a constant?

Reply:
Sorry, the other piece to that: I think part of what I don't like with `is_env_var` is that it suggests it is *always* an env var - but I'm imagining plenty of things that might be loadable from the environment without having to be.

Would `is_env_loadable` make everyone happy?

Reply:
Whilst we're on naming, I was musing on my cycle to work today whether `fully_qualified_name` would actually be better than `global_name` .

I'm worried about potential misidentification with `$_GLOBAL` variables, which is a whole nother thing...